ReactNative/ReactTwitter/core/views/debug/debug-screen.js

```diff
@@ -3,59 +3,88 @@
 import React from 'react'
 import {
   StyleSheet,
-  View,
-  Navigator
+  Navigator,
+  ListView
 } from 'react-native'
 
 var Button = require('react-native-button')
 
+var DeepLinkingView = require('./deep-linking.js')
+var PushNotificationsScreenView = require('./push-notifications-screen.js')
+var OauthView = require('./oauth-screen.js')
+var DebugAnimationScene = require('./debug-animation-scene.js')
+var DebugNativeComponentScene = require('./debug-native-component-scene.js')
+
+const scenes = [
+  DeepLinkingView,
+  PushNotificationsScreenView,
+  OauthView,
+  DebugAnimationScene,
+  DebugNativeComponentScene
+]
+
 var DebugScreenView = React.createClass({
+  statics: {
+    navigatorID () {
+      return 'debug-screen-identifier'
+    },
+    navigatorTitle () {
+      return 'Debug Screen'
+    }
+  },
   propTypes: {
     navigator: React.PropTypes.instanceOf(Navigator).isRequired
   },
 
+  getInitialState: function () {
+    var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})
+    let rows = scenes.map((scene) => {
+      return {
+        name: scene.navigatorTitle(),
+        sceneID: scene.navigatorID()
```

Comment:
I don't think it's possible yet.
If you need to be this powerful I strongly advise you to move to TypeScript, which [has interfaces](https://www.typescriptlang.org/docs/handbook/interfaces.html) and is a language on its own.

To circumvent the limitation, you can create your own `StatefulClassCreator` with a `.createClass` method that checks that the passed object has the `getInitialState` property and, if so, delegates it to `React.createClass`.